### ISSUE TITLE
No validation dataset

### ISSUE THREAD

**Issue by pamparana34** [PyTorchLightning/pytorch-lightning#2063]
I am just starting to use pytorch_lightning. I have one question regarding the validation set. I may or may not have a validation set during my training. How should the `validation_step` be structured in this case?

is it enough do something like this:

```
@pl.data_loader
def val_dataloader(self):
     if has_valset:        
        return self.__dataloader(train=False)
    return None
```

So if it  returns None, then no validation loop calls will be made?

Or do I need something like:

```
def validation_step(self, batch, batch_idx):
    # Is this check necessary?
    if not has_vaset:
        return
   
   # process the validation batch
```

**Comment by awaelchli** [comment]
I suggest you return the validation loader anyway, and then set the Trainer flag val_percent_check=0 if you don't want to run validation. 
Let me know if that works or not.

**Comment by Borda** [comment]
feel free to reopen if needed 🐰
